### PR TITLE
docs(community): Improves community standards

### LINKED ISSUE #255: Add a CONTRIBUTING.md (or asciidoc)
The document should have guidelines on how to contribute to this repository (commit messages, code format, etc)

### PR DESCRIPTION
**Pull request by ia3andy**
This PR includes 5 new documents:
    - CONTRIBUTING.md
    - CODE_OF_CONDUCT.md
    - COMMIT_MSG.md
    - PULL_REQUEST_TEMPLATE.md
    - ISSUE_TEMPLATE.md

with links in the README.md

This is a first shot and is meant to be discussed & evolve.

Closes #255

### CODE REVIEW

**Comment by ia3andy** [comment]
@ALRubinger @quintesse @gastaldi @edewit:
I am still working on it, but I make it available so you we can start discussing things together.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+### Prerequisites
+
+* Java 8+
+* Maven 3.5.3+
+* Git
+
+### Git Workflow
+
+To contribute to Forge, fork this repository to your own Git, clone your fork, commit your work on topic branches, and make pull requests.
```
Comment (gastaldi) [inline CONTRIBUTING.md]: To Forge?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+Read [this document](./README.md).
+
+## Choosing issues to work on
+
+If you're wondering what issues would be suitable when you're just getting started, we recommend taking a look at the [issues](https://github.com/fabric8-launcher/launcher-backend/issues) with the '[good first issue](https://github.com/fabric8-launcher/launcher-backend/labels/good%20first%20issue)' and '[help wanted](https://github.com/fabric8-launcher/launcher-backend/labels/help%20wanted)' labels.
+
+## License Information and Contributor Agreement
+
+* There is no need to sign a contributor agreement to contribute to JBoss Forge. You just need to explicitly license any contribution under the EPL 1.0. If you add any new files to JBoss Forge, make sure to add the correct header.
```
Comment (gastaldi) [inline CONTRIBUTING.md]: JBoss Forge?

File: COMMIT_MSG.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+- core
+- web
+- osio-addon
+- git
+- keycloak
+- etc.
+
+The `<scope>` can be empty (e.g. if the change is a global or difficult to assign to a single component), in which case the parentheses are omitted.
+
+## Message body
```
Comment (quintesse) [inline COMMIT_MSG.md]: I think we talked about this and decided that we don't really care about this part. There are many foreigners with only limited English language skills. Let them write how they feel comfortable.

File: COMMIT_MSG.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+- uses the imperative, present tense: “change” not “changed” nor “changes”
+- includes motivation for the change and contrasts with previous behavior
+
+For more info about message body, see:
+- http://365git.tumblr.com/post/3308646748/writing-git-commit-messages
+- http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
+
+## Message footer
```
Comment (quintesse) [inline COMMIT_MSG.md]: Missing info here?

**Comment by edewit** [comment]
do we really need a code of conduct that's so formal

File: COMMIT_MSG.md (hunk 4 above)
Comment (quintesse) [reply COMMIT_MSG.md]: Ah the other items are sub-items of this.
Then I'd probably add a simple statement like "In the footer you should mention issues and any breaking changes"

File: COMMIT_MSG.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+or in the case of multiple issues:
+
+```
+Closes #123, #245, #992
+```
+
+### Breaking changes
+
+All breaking changes have to be mentioned in footer with the description of the change, justification and migration notes.
+
```
Comment (quintesse) [inline COMMIT_MSG.md]: I'd explicitly put "For example:" here.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+To contribute to the Fabric8-Launcher Backend:
+
+1. Fork this repository to your own Git.
+2. Clone your fork.
+3. Commit/Push your work on topic branches
+4. Make pull requests.
+
+Here are 3 simple rules we follow:
+- Use a fork of our repository with a topic branch to contribute.
+- Redact a beautiful commit/pull-request message following our [guide](./COMMIT_MSG.md).
```
Comment (quintesse) [inline CONTRIBUTING.md]: I'd use "nice" instead of "beautiful", it sounds very lyrical/poetical to me now ;)

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+- Try to keep a clean commit log (no unnecessary commits, no merge commits).
+
+In order to make you feel safe regarding your contribution, we are here to help you:
+ - Unit/Integration tests and checkstyle are run for every modification you make.
+ - There is a test keeper plugin so you don't forget to test your changes if necessary.
```
Comment (quintesse) [inline CONTRIBUTING.md]: Is this line necessary? It seems very specific.
(And I'm already getting annoyed by this stupid plugin, so let's see how long it survives :-p )

Comment (ia3andy) [reply CONTRIBUTING.md]: @quintesse lol I was so sure you would get annoyed by the plugin that I want to keep this line just for you :)

File: COMMIT_MSG.md (hunk 3 above)
Comment (ia3andy) [reply COMMIT_MSG.md]: what do you mean?

Comment (quintesse) [reply COMMIT_MSG.md]: What do _you_ mean? Or better, what is it you don't understand?

Comment (ia3andy) [reply COMMIT_MSG.md]: loool :)

what part do you mean? Or better could you rephrase the way you see it?

Comment (quintesse) [reply COMMIT_MSG.md]: Ha, don't look at the "diff" GitHub is showing you , buit the _line_ I put the comment on hehe

Comment (quintesse) [reply COMMIT_MSG.md]: I'm talking about the "message body"

**Comment by edewit** [comment]
:+1:
